docs/source/tutorials_source/package/tutorial_custom_augmentations.py

```diff
@@ -0,0 +1,366 @@
+#
+# The choice of the optimizer is left to the user. Here, we go with simple stochastic
+# gradient descent with momentum.
+criterion = lightly.loss.NTXentLoss(
+    temperature=0.1,
+    memory_bank_size=4096,
+)
+
+optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9, weight_decay=1e-4)
+
+
+# %%
+# Train MoCo with custom augmentations
+# -------------------------------------
+# Let's put all the puzzle pieces together and combine them in an encoder object.
+# Training the self-supervised model is now very easy. Simply call the function
+# `train_embedding` with the desired arguments.
+
+encoder = lightly.embedding.SelfSupervisedEmbedding(
+    model,
+    criterion,
+    optimizer,
+    dataloader_train,
+)
+
+gpus = 1 if torch.cuda.is_available() else 0
+encoder.train_embedding(gpus=gpus,
+                        progress_bar_refresh_rate=100,
+                        max_epochs=max_epochs,
+                        precision=16)
+
+
+# %%
+# Evaluate the results
+# ------------------------
+# It's always a good idea to evaluate how good the learned representations really
+# are. How to do this depends on the available data and metdata. Luckily, in our case,
+# we have annotations of critical findings on the X-ray images. We can use this information
+# to see, whether images with similar annotations are grouped together.
+#
+# We start by getting a vector representation of each image in the dataset. For this,
+# we create a new dataloader. This time, we can pass the transform directly to the dataset.
+
+# test transforms differ from training transforms as they do not introduce
+# additional noise
+test_transforms = torchvision.transforms.Compose([
+    HistogramNormalize(),
+    torchvision.transforms.Grayscale(num_output_channels=3),
+    torchvision.transforms.Resize(input_size),
+    torchvision.transforms.ToTensor(),
+])
+
+# create the dataset and overwrite the image loader as before
+dataset_test = lightly.data.LightlyDataset(
+    input_dir=path_to_data,
+    transform=test_transforms
+)
+dataset_test.dataset.loader = tiff_loader
+
+# create the test dataloader
+dataloader_test = torch.utils.data.DataLoader(
+    dataset_test,
+    batch_size=1,
+    shuffle=False,
+    drop_last=False,
+    num_workers=num_workers
+)
+
+# get vector representations for all images in the dataset
+encoder.to('cuda')
+embeddings, _, fnames = encoder.embed(dataloader_test, device='cuda')
+embeddings = normalize(embeddings)
+
+# %%
+# Now, we can use the embeddings to search for nearest neighbors.
+#
+# We choose three example images. For each example image, we find 50 nearest neighbors.
+# Then, we plot the critical findings in the example image (dark blue) and the distribution
+# of the critical findings in the nearest neighbor images (light blue) as bar plots.
+
+
+# transform the original annotations to multiclass labels
+fnames = [fname.split('.')[0] for fname in fnames]
+
+df = pandas.read_csv('/datasets/vinbigdata/train.csv')
```

Comment:
This block is a bit too magic. Maybe show the first 3-4 rows of the csv file (including heading) in the comments/ text. So someone just reading/ skimming through the tutorial would get an idea why we have the next few lines?

Reply:
Maybe also mention what the output is. Show an example of what is in `multilabels` after the "magic" :)

Reply:
I think I'll just add an explanation, the CSV file is not so informative.